src/main/java/software/amazon/cloudwatchlogs/emf/environment/AgentBasedEnvironment.java

```diff
@@ -45,7 +45,17 @@ public String getName() {
 
     @Override
     public String getLogGroupName() {
-        return config.getLogGroupName().orElse(getName() + "-metrics");
+        if (config.getLogGroupName().isPresent()) {
+            return config.getLogGroupName().get();
+        } else {
+            String serviceName = getName();
+            // for ECS services, replace "repo:tag" format with "repo-tag" to satisfy
+            // log group regex
+            if (serviceName.indexOf(':') != -1) {
+                serviceName = serviceName.replaceAll(":", "-");
+            }
```

Comment:
Is this line necessary? String.replaceAll() won't do anything if a colon isn't present.
```suggestion
            serviceName = serviceName.replaceAll(":", "-");
```

Reply:
Solved.